cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs

```diff
@@ -190,20 +196,32 @@ pub type PoolFungiblesTransactor = FungiblesAdapter<
 pub type AssetTransactors =
 	(CurrencyTransactor, FungiblesTransactor, ForeignFungiblesTransactor, PoolFungiblesTransactor);
 
-/// Simple `MultiLocation` matcher for Local and Foreign asset `MultiLocation`.
-pub struct LocalAndForeignAssetsMultiLocationMatcher;
-impl MatchesLocalAndForeignAssetsMultiLocation for LocalAndForeignAssetsMultiLocationMatcher {
-	fn is_local(location: &MultiLocation) -> bool {
-		use assets_common::fungible_conversion::MatchesMultiLocation;
-		TrustBackedAssetsConvertedConcreteId::contains(location)
+/// Simple `Location` matcher for Local and Foreign asset `Location`.
+pub struct LocalAndForeignAssetsLocationMatcher;
+impl MatchesLocalAndForeignAssetsLocation<xcm::v3::Location>
```

Comment:
Can you describe how the XCM upgrade should happen? I see the runtimes use a mix of v4 and v3 - for example, here, this matcher is only implemented for v3.

I would prefer a PR that adds v4 then (later) one that switches runtimes to using it - doing both in the same PR is also ok, but this current PR only does the "upgrade" partially IIUC..

Reply:
It would've been easier to break it down into two PRs, one for introducing the new version and another to change to it. Some additional tools would be welcome to make this process smoother, which I'll look into when introducing newer versions. This PR does the whole upgrade, all the v3 stuff is just for storage items which were already v3 instead of `Versioned*`. We are going to want to migrate all of them after this PR.